### PR TITLE
Use _REMOTE_USER when available

### PR DESCRIPTION
**Pull request by Chuxel**
This PR updates features to take advantage of the new _REMOTE_USER environment variable. It leaves the existing logic in place to handle older versions of the dev container CLI (and to try to be smart in the case when no remote user is set).

Priority is:
1. `USERNAME` (if this was added as an option)
2. `_REMOTE_USER`
3. `automatic` - this is the final fallback to try to auto-detect and for back compat.

The PR also makes sure we are quoting inbound feature options to avoid spaces, newlines, etc causing problems.

### CODE REVIEW

File: src/python/install.sh [hunk 1]
```diff
@@ -7,24 +7,24 @@
...
 # Maintainer: The VS Code and Codespaces Teams
 
-PYTHON_VERSION=${VERSION:-"latest"} # 'system' checks the base image first, else installs 'latest'
-INSTALL_PYTHON_TOOLS=${INSTALLTOOLS:-"true"}
-OPTIMIZE_BUILD_FROM_SOURCE=${OPTIMIZE:-"false"}
-PYTHON_INSTALL_PATH=${INSTALLPATH:-"/usr/local/python"}
-OVERRIDE_DEFAULT_VERSION=${OVERRIDEDEFAULTVERSION:-"true"}
+PYTHON_VERSION="${VERSION:-"latest"}" # 'system' or 'os-provided' checks the base image first, else installs 'latest'
+INSTALL_PYTHON_TOOLS="${INSTALLTOOLS:-"true"}"
+OPTIMIZE_BUILD_FROM_SOURCE="${OPTIMIZE:-"false"}
```
Comment (samruddhikhandale) [inline src/python/install.sh:12]:
```suggestion
OPTIMIZE_BUILD_FROM_SOURCE="${OPTIMIZE:-"false"}"
```

I wonder if the Python test failures are due to missing closing quote.

File: src/python/install.sh [hunk 2]
```diff
@@ -7,24 +7,24 @@
...
+PYTHON_INSTALL_PATH="${INSTALLPATH:-"/usr/local/python"}
```
Comment (samruddhikhandale) [inline src/python/install.sh:13]:
```suggestion
PYTHON_INSTALL_PATH="${INSTALLPATH:-"/usr/local/python"}"
```

**Review by samruddhikhandale — COMMENTED** [review]
This is awesome. Thank you for the PR. ✨ 

Added a comment which should help fix the Python tests.
The failing oryx tests  (jammy) are expected.

File: src/anaconda/install.sh [hunk 3]
```diff
@@ -8,10 +8,10 @@
 # Maintainer: The VS Code and Codespaces Teams
 
 
-VERSION=${VERSION:-"latest"}
-USERNAME=${USERNAME:-"automatic"}
-UPDATE_RC=${UPDATE_RC:-"true"}
-CONDA_DIR=${CONDA_DIR:-"/usr/local/conda"}
+VERSION="${VERSION:-"latest"}"
```
Comment (samruddhikhandale) [inline src/anaconda/install.sh:11]: ℹ️ Can we bump up patch versions for all the Features in respective `devcontainer-feature.json` files? 
The Features will be automatically published once this PR is merged. (Provided the version has changed)

Comment (Chuxel) [reply src/anaconda/install.sh:11]: Done!

**Comment by Chuxel** [comment]
@samruddhikhandale @joshspicer Adapted to feedback.

**Comment by samruddhikhandale** [comment]
Oops, I didn't realize this existed and created a duplicate https://github.com/devcontainers/features/pull/309 🤦‍♀️ 
This one does few other things as well, closing mine & reviewing this one now 👀

File: src/php/install.sh [hunk 4]
```diff
@@ -11,17 +11,17 @@ set -eux
...
-export PHP_DIR=${PHP_DIR:-"/usr/local/php"}
-USERNAME=${USERNAME:-"automatic"}
-UPDATE_RC=${UPDATE_RC:-"true"}
+export PHP_DIR="${PHP_DIR:-"/usr/local/php"}"
+USERNAME="${USERNAME:-"${_REMOTE_USER:-"automatic"}"}"
+UPDATE_RC="${UPDATE_RC:-"true"}"
 
 # Comma-separated list of php versions to be installed
 # alongside VERSION, but not set as default.
-ADDITIONAL_VERSIONS=${ADDITIONALVERSIONS:-""}
```
Comment (samruddhikhandale) [inline src/php/install.sh:24]: https://github.com/devcontainers/features/pull/309/commits/627fc411904a0a68d7e3d9cc6a7af79609d57720 Making similar changes should fix failing the php:ubuntu:bionic test.

Comment (samruddhikhandale) [reply src/php/install.sh:24]: We could fix the test in another PR 👍

**Review by samruddhikhandale — APPROVED** [review]

File: src/php/install.sh (hunk 4 above)
Comment (Chuxel) [reply src/php/install.sh:24]: Yeah, and the Java one has this error in it for `install_from_non_default_distro`:

```
#16 0.950 find: Failed to restore initial working directory: /tmp/build-features/java_4: Permission denied
#16 0.961 find: Failed to restore initial working directory: /tmp/build-features/java_4: Permission denied
#16 0.963 find: Failed to restore initial working directory: /tmp/build-features/java_4: Permission denied
```

Which again seems like a general bug.

**Pull request merged**